### ISSUE TITLE
[FEATURE] Descriptive representational strings for params

### ISSUE THREAD

**Issue by mRokita** [tiangolo/fastapi#1398]
### Is your feature request related to a problem

Kind of, recently I generated a documentation for one of my personal projects using [mkdocstrings](https://pawamoy.github.io/mkdocstrings/) and that's how it renders params:
![](https://i.imgur.com/znm6F8z.png)

That's the behavior I would prefer:

![](https://i.imgur.com/28wNWOL.png)

### The solution you would like

Implement `__repr__` for all classes in `fastapi.params`.
I am willing to do that, but first I wanted to make sure that this is a good idea and such PR would be accepted.

### Describe alternatives you've considered

A dirty hack like this:

```python
from fastapi.params import Depends as Dep

class Depends(Dep):

    def __init__(self, dependency = None, *, use_cache=True):
        super().__init__(dependency, use_cache=True)

    def __repr__(self) -> str:
        name = getattr(self.dependency, "__name__", type(self.dependency).__name__)
        args = name
        if not self.use_cache:
            args += ", use_cache=False"
        return f"Depends({args})"

```

Works, but it would be nice to have it by default.

**Comment by tiangolo** [comment]
Ah, interesting, yeah, I would accept a PR doing that and testing it :heavy_check_mark: :nerd_face:

**Comment by fabrizzioalco** [comment]
Hey is this done already? I would like to contribute

**Comment by tiangolo** [comment]
Not done yet, you can go ahead @fabrizzioalco  :nerd_face: :rocket:

**Comment by juped** [comment]
I think this would be a good issue for @victorphoenix3 and @rkbeatss to work on - I don't want to "steal" it out from under @fabrizzioalco if he's already started, though.

**Comment by tiangolo** [comment]
Yep, there's no PR with this yet, so if @victorphoenix3 or @rkbeatss are ready, go ahead! :rocket: :nerd_face:

**Comment by tiangolo** [comment]
This was implemented by @rkbeatss and @victorphoenix3 in https://github.com/tiangolo/fastapi/pull/1560 :rocket:

It should be available from FastAPI version `0.57.0` :tada:
